packages/viz/src/lib/sources/CARTOSource.ts

```diff
@@ -95,10 +96,108 @@ export class CARTOSource extends Source {
     return this._credentials;
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
-  // eslint-disable-next-line
-  public async getMetadataForField(field: string): Promise<Metadata> {
-    throw new Error('Method not implemented.');
+  public getFieldStats(): NumericFieldStats | CategoryFieldStats {
+    // initialize the stats to 0
+
+    if (!this.isInitialize)
+      throw new Error('To run this method you need to first call to init()');
+
+    if (this._fieldStats === undefined) {
+      throw new Error('Stats are not set after map instantiation');
+    }
+
+    return this._fieldStats;
+  }
+
+  private _initConfigForStats(field: string) {
+    if (this._mapConfig.metadata === undefined) {
+      throw new Error('Map Config has not metadata fiedl');
+    }
+
+    // Modify mapConfig to add the field stats
+    this._mapConfig.metadata.columnStats = {
+      topCategories: 32768,
+      includeNulls: true
+    };
+    this._mapConfig.metadata.dimensions = true;
+
+    /* eslint-disable @typescript-eslint/camelcase */
+    this._mapConfig.metadata.sample = {
+      num_rows: 1000,
```

Comment:
In VL we made this  `num_rows` a param, as it was pretty relevant for histograms. But we can leave it as it is for now, and extend later

Reply:
Done